### PR TITLE
Validate filename vs JSON hierarchy

### PR DESCRIPTION
**Pull request by gsouquet**
Hi,

After doing a few copy paste I realised that one thing was not validated by the linter.

Given a file with the filename: `AbortController.json` we expect a JSON hierarchy like the following

```
{
  "api": {
    "AbortController": {}
  }
}
```

However if a typo in the `AbortController` won't be caught by the JSON schema as this value is different for every file. I decided to add a validation step for this to ensure a quicker feedback loop.

Given that the user made a typo in the interface name, `npm run show-errors` would output

```
Incorrect hierarachy in file: api/AbortController.json
Make sure the filename is correct and matches the interface name
Expected hierarchy based on the current file name: api > AbortController
```

This PR has highlighted two issues, the `http` folder is the only one where the capitalization in the JSON hierarchy is different than in the filename.

And the file `globals.json` is the only one that would wrap the builtins object in a higher `globals` parent.

I went ahead and fixed `globals.json` and `sample-data.json` to fix the linter errros, however I did not change anything in the `http` folders.

I'm keen to hear your thoughts on that, but I think it can help prevent errorneous code to be pushed.

### CODE REVIEW

File: test/lint.js [hunk 1]
```diff
@@ -17,11 +18,12 @@ function load(...files) {
         if (file.indexOf('browsers/') !== -1) {
           hasSchemaErrors = testSchema(file, './../schemas/browsers.schema.json');
         } else {
-          hasSchemaErrors = testSchema(file);
-          hasStyleErrors = testStyle(file);
-          hasVersionErrors =  testVersions(file);
+          // hasSchemaErrors = testSchema(file);
+          // hasStyleErrors = testStyle(file);
+          // hasVersionErrors =  testVersions(file);
```
Comment (ExE-Boss) [inline test/lint.js]: These probably shouldn’t be commented out.

File: test/test-hierarchy.js [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+
+  var data = require(dataFilename);
+
+  var localFilePath = dataFilename.replace(path.resolve(__dirname, '..') + path.sep, '');
+  var levels = localFilePath.replace(".json", "").split("/");
+
+  var hasErrors = false;
+
+  // HTTP is the only folder that have non matching capitalization
+  // between the file name and the compat data content.
```
Comment (ExE-Boss) [inline test/test-hierarchy.js]: The comment should be:
```js
  // HTTP is the only folder that has non matching capitalization
  // between the file name and the compat data content.
```

**Comment by gsouquet** [comment]
Thanks for the review @ExE-Boss .
I did fix everything. Happy to discuss the changes I made anytime. Let me know if you need anything from me.

**Comment by Elchi3** [comment]
Oh and please rebase against master, there has been many more files added in the meantime. I hope al is right with those, but you never know without tests :)

**Comment by gsouquet** [comment]
Thanks for the comments @Elchi3 .

Don't worry, I believe the MDN hack week-end in Paris added a lot more work for you ;) I haven't been very available in the past few weeks either. 

Will look into that this week 👍

**Comment by gsouquet** [comment]
Ahhh... 
I'm not sure what happened here. Looks like my rebase went completely wrong

**Comment by jwhitlock** [comment]
I took a look at trying to fix the rebase, and I doubt it is worth the effort.

I think you should start a new branch off the current master, apply your changes to ``test/lint.js``, and then fix the current JSON. You can submit a new PR and link to this one.  Otherwise, you are stuck manually merging JSON changes, and that's robot work.

**Comment by Elchi3** [comment]
Closing here as we can't proceed with this PR. 
I think it would be great if @ExE-Boss opens a new PR given there is https://github.com/ExE-Boss/mdn-browser-compat-data/tree/gsouquet/linter-api-name that contains this work. Or, of course, I would also love to see a new PR from @gsouquet.
Thanks all.

**Pull request closed without merging**